File: src/main/java/net/atos/zac/event/Opcode.java
/*
 * SPDX-FileCopyrightText: 2021 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.event;

import java.util.EnumSet;
import java.util.Set;

/**
 * Enumeration containing the operations as used by the {@link AbstractEvent}.
 * Maps to opcode.ts
 */
public enum Opcode {

    /**
     * indication that the mentioned object has been added
     */
    CREATED,

    /**
     * indication that the mentioned object has been updated
     */
    UPDATED,

    /**
     * indication that the mentioned object has been deleted
     */
    DELETED,

    /**
     * indication that something has happened to the mentioned object
     */
    ANY;

    public static Set<Opcode> any() {
        return EnumSet.complementOf(EnumSet.of(ANY));
    }
}

File: src/main/app/src/app/core/websocket/model/opcode.ts
/*
 * SPDX-FileCopyrightText: 2021 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

/**
 * Maps to OpcodeEnum.java
 *
 * Note that the CREATED value is not available for subscription (reason: new objectIds will be unknown client side ;-)
 */
export enum Opcode {
  UPDATED = "UPDATED",
  DELETED = "DELETED",
  ANY = "ANY",
}
